protocol/test/forge/v2/interestRate/LinearWithKinkInterestRateModel.t.sol

```diff
@@ -0,0 +1,158 @@
+  ) internal pure returns (uint256) {
+    // Approxmiate as P * (1 + r/365 days)^(age)
+    // ie compounding every 1 second (almost but not quite continuous)
+    uint256 onePlusRate = uint256(rate / 365 days + 1e18);
+
+    return ud(principal).mul(ud(onePlusRate).powu(age)).unwrap();
+  }
+}
+
+contract LinearWithKinkInterestRateModelTestModifiers is
+  LinearWithKinkInterestRateModelTestBase
+{
+  event InterestRateParamsSet(
+    uint256 _baseInterestRate,
+    uint256 _maxInterestRate,
+    uint256 _kinkUtilizationRatio,
+    uint256 _kinkInterestRate
+  );
+
+  using SafeCast for uint256;
+
+  function test_accessSetRateParams() public {
+    // no access for alice
+    {
+      vm.startPrank(alice);
+      vm.expectRevert(
+        abi.encodeWithSelector(CommonEventsAndErrors.InvalidAccess.selector)
+      );
+      daiInterestRateModel.setRateParams(0.02e18, 0.75e18, 0.5e18, 0.1e18);
+      vm.stopPrank();
+    }
+
+    // access for executor
+    {
+      vm.startPrank(executor);
+      vm.expectEmit();
+      emit InterestRateParamsSet(0.02e18, 0.75e18, 0.5e18, 0.1e18);
+      daiInterestRateModel.setRateParams(0.02e18, 0.75e18, 0.5e18, 0.1e18);
+      vm.stopPrank();
+    }
+
+    // access for rescuer
+    {
+      vm.startPrank(rescuer);
+      vm.expectRevert(
+        abi.encodeWithSelector(CommonEventsAndErrors.InvalidAccess.selector)
+      );
+      daiInterestRateModel.setRateParams(0.02e18, 0.75e18, 0.5e18, 0.1e18);
+
+      daiInterestRateModel.setRescueMode(true);
+      vm.expectEmit();
+      emit InterestRateParamsSet(0.02e18, 0.75e18, 0.5e18, 0.1e18);
+      daiInterestRateModel.setRateParams(0.02e18, 0.75e18, 0.5e18, 0.1e18);
+      vm.stopPrank();
+    }
+  }
+
+  function test_getRateParams() public {    
+    uint80 testBaseInterestRate = uint256(0.05e18).encodeUInt80(); // 5%
+    uint80 testMaxInterestRate = uint256(0.2e18).encodeUInt80(); // 20%
+    uint80 testKinkInterestRate = uint256(0.1e18).encodeUInt80(); // 10%
+    uint256 testKinkUtilizationRatio = uint256(0.9e18); // %90%
+    
+    // get initial params
+    (
+      uint256 baseInterestRate,
+      uint256 maxInterestRate,
+      uint256 kinkInterestRate,
+      uint256 kinkUtilizationRatio
+    ) = daiInterestRateModel.rateParams();
+    
+    assertEq(baseInterestRate, testBaseInterestRate);
+    assertEq(maxInterestRate, testMaxInterestRate);
+    assertEq(kinkInterestRate, testKinkInterestRate);
+    assertEq(kinkUtilizationRatio, testKinkUtilizationRatio);
+  }
+
+  function test_utilizationRation() public {
+    assertEq(utilizationRatioPct, 0.5e18); // 50%
+  }
+  function test_expectedInterestRate() public {
+    assertEq(expectedDaiInterestRate, 0.077777777777777777e18); // ~7%
+  }
+  function test_expectedDebtAmount() public {
+    assertEq(expectedDaiDebt, 54_044.121788100384500000e18); // ~54k
+  }
+  function test_expectedUtilizationRatio() public {
+    assertEq(expectedUtilizationRatio, 0.540441217881003845e18); // ~54%
+  }
+}
```

Comment:
- Coverage 100% for `LinearWithKinkInterestRateModel.sol`
- Coverage still not 100 for `BaseInterestRateModel.sol`, it may be because it's an abstract contract
- Coverage not working for `CompoundedInterest.sol`, prob because is a library?

<img width="1017" alt="image" src="https://github.com/TempleDAO/temple/assets/127277105/958d1591-9ad9-4aa4-93fd-251d4ce18032">